internal/build/helper.go

```diff
@@ -44,22 +44,22 @@ func (m *helper) ApplyBuildArgOverrides(args []kmmv1beta1.BuildArg, overrides ..
 	return args
 }
 
-func (m *helper) GetRelevantBuild(modSpec kmmv1beta1.ModuleSpec, km kmmv1beta1.KernelMapping) *kmmv1beta1.Build {
-	if modSpec.ModuleLoader.Container.Build == nil {
-		return km.Build.DeepCopy()
+func (m *helper) GetRelevantBuild(moduleBuild *kmmv1beta1.Build, mappingBuild *kmmv1beta1.Build) *kmmv1beta1.Build {
+	if moduleBuild == nil {
+		return mappingBuild.DeepCopy()
 	}
 
-	if km.Build == nil {
-		return modSpec.ModuleLoader.Container.Build.DeepCopy()
+	if mappingBuild == nil {
+		return moduleBuild.DeepCopy()
 	}
 
-	buildConfig := modSpec.ModuleLoader.Container.Build.DeepCopy()
-	buildConfig.DockerfileConfigMap = km.Build.DockerfileConfigMap
+	buildConfig := moduleBuild.DeepCopy()
+	buildConfig.DockerfileConfigMap = mappingBuild.DockerfileConfigMap
```

Comment:
What if the mapping's `DockerfileConfigMap` is nil and the `Module` isn't? in that situation, I think we should use the `Module`'s.

Reply:
Nice catch! Fixed. BTW, this is one of the reason that i think mergo will be problematic: much easier to debug this cases in the code then in mergo' tags.